server/src/main/java/org/apache/druid/client/CachingClusteredClient.java

```diff
@@ -473,6 +473,9 @@ private Set<SegmentServerSelector> computeSegmentsToQuery(
         }
         for (PartitionChunk<ServerSelector> chunk : filteredChunks) {
           ServerSelector server = chunk.getObject();
+          if (query.context().isRealtimeSegmentsOnly() && !server.isRealtimeSegment()) {
```

Comment:
Evaluate and assign `query.context().isRealtimeSegmentsOnly()` once outside the for loops for reuse